client/src/pages/Booking/BookingSlot/BookingSlot.tsx

```diff
@@ -0,0 +1,91 @@
+import { Box, Button, Card, CardContent, Grid, Menu, MenuItem, Typography } from '@mui/material';
+import SettingsIcon from '@mui/icons-material/Settings';
+import { BookingInfo } from '../../../interface/BookingInfo';
+import AvatarDisplay from '../../../components/AvatarDisplay/AvatarDisplay';
+import { useStyles } from './useStyles';
+import React, { Key, SyntheticEvent, useState } from 'react';
+import { updateBooking } from '../../../helpers/APICalls/bookingInfo';
+
+interface Props {
+  bookingInfo: BookingInfo;
+  key: Key | null | undefined;
+  updatable: boolean;
+}
+
+const BookingSlot = (props: Props): JSX.Element => {
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const openSetting = Boolean(anchorEl);
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const updateBookingInfo = (event: SyntheticEvent) => {
+    if (event.currentTarget.textContent === 'Accept') {
+      props.bookingInfo.status = 'accepted';
+      updateBooking(props.bookingInfo._id, { status: 'accepted' });
```

Comment:
We could also update the snackbar once we've successfully update the status of a booking (here and below)